eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -284,65 +332,299 @@ function Invoke-Generate() {
 }
 
 function Invoke-Build() {
     param(
         [string]$sdkfolder= ""
     )
     $sdkfolder = $sdkfolder -replace "\\", "/"
-    Push-Location $sdkfolder
-    dotnet build
+    dotnet build $sdkfolder
     if ( !$? ) {
         Write-Error "Failed to build sdk. exit code: $?"
-        Pop-Location
         exit 1
     }
-    Pop-Location
 }
 
 function Invoke-Pack() {
     param(
         [string]$sdkfolder= ""
     )
     $sdkfolder = $sdkfolder -replace "\\", "/"
-    Push-Location $sdkfolder
-    dotnet pack
+    dotnet pack $sdkfolder /p:RunApiCompat=$false
     if ( !$? ) {
         Write-Error "Failed to build sdk package. exit code: $?"
-        Pop-Location
         exit 1
     }
-    Pop-Location
 }
 function Get-ResourceProviderFromReadme($readmeFile) {
-    $readmeFileRegex = "(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    $readmeFileRegexWithSpec = "specification/(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    try
-    {
-        if ($readmeFile -match $readmeFileRegexWithSpec)
-        {
-            return $matches["specName"], $matches["serviceType"]
+    $readmeFile = $readmeFile -replace "\\", "/"
+    $pathArray = $readmeFile.Split("/");
+
+    if ( $pathArray.Count -lt 3) {
+        Write-Error "Error: invalid readme file path."
+        exit 1
+    }
+
+    $specName = $pathArray[-3]
+    $serviceType = $pathArray[-2]
+    Write-Host "specName:$specName, serviceType: $serviceType"
+
+    return $specName, $serviceType
+}
+
+function Invoke-GenerateAndBuildSDK () {
+    param(
+        [string]$readmeAbsolutePath,
+        [string]$sdkRootPath,
+        [string]$autorestConfigYaml = "",
+        [string]$downloadUrlPrefix = "",
+        [object]$generatedSDKPackages
+    )
+    $readmeFile = $readmeAbsolutePath -replace "\\", "/"
+    Write-Host "readmeFile:$readmeFile"
+    $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
+    Write-Host "service:$service, serviceType:$serviceType"
+    
+    if (!$readmeFile.StartsWith("http") -And !(Test-Path -Path $readmeFile)) {
+        Write-Error "readme file does not exist."
+        exit 1
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
+        if ($packageNameHash[$service] -ne "") {
+            $package = $packageNameHash[$service]
         }
-        if ($readmeFile -match $readmeFileRegex)
-        {
-            return $matches["specName"], $matches["serviceType"]
+        New-MgmtPackageFolder -service $service -packageName $package -sdkPath $sdkRootPath -commitid $commitid -readme $readmeFile -outputJsonFile $newpackageoutput
+        if ( !$?) {
+            Write-Error "Failed to create sdk project folder. exit code: $?"
+            exit 1
+        }
+        $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
```

Comment:
Please use camel case for variable names for readability.

Reply:
updated